evennia/objects/objects.py

```diff
@@ -1249,7 +1249,7 @@ def get_numbered_name(self, count, looker, **kwargs):
 
         """
         plural_category = "plural_key"
-        key = kwargs.get("key", self.key)
+        key = kwargs.get("key", self.name)
```

Comment:
This change needs to be mentioned on line 1240 too - explaining that if `key` is not given, the `.name` property will be used instead.